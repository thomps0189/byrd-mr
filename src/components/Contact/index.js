import React, { useState } from "react";
import { validateEmail } from "../../utils/helpers";

function Contact() {
  const [formState, setFormState] = useState({
    name: "",
    email: "",
    message: "",
  });

  const [errorMessage, setErrorMessage] = useState("");
  const { name, email, message } = formState;

  const handleSubmit = (e) => {
    e.preventDefault();
    if (!errorMessage) {
      console.log("Submit Form", formState);
    }
  };

  const handleChange = (e) => {
    if (e.target.name === "email") {
      const isValid = validateEmail(e.target.value);
      if (!isValid) {
        setErrorMessage("Your email is invalid.");
      } else {
        setErrorMessage("");
      }
    } else {
      if (!e.target.value.length) {
        setErrorMessage(`${e.target.name} is required.`);
      } else {
        setErrorMessage("");
      }
    }
    if (!errorMessage) {
      setFormState({ ...formState, [e.target.name]: e.target.value });
      console.log("Handle Form", formState);
    }
  };
  return (
    <section id="form-container">
      <form id="contact-form" onSubmit={handleSubmit}>
        <label id="form-header" for="full-name">
          Full Name
        </label>
        <input
          type="text"
          id="full-name"
          name="full-name"
          defaultValue={name}
          placeholder="Enter your full name"
          onBlur={handleChange}
        />
        <label id="form-header" for="email">
          Email
        </label>
        <input
          type="email"
          id="email"
          name="email"
          defaultValue={email}
          placeholder="Enter your email"
          onBlur={handleChange}
        />
        <label id="form-header" for="message">
          Message
        </label>
        <input
          type="text"
          id="message"
          name="message"
          placeholder="What are your plans?"
          defaultValue={message}
          onBlur={handleChange}
        />
      </form>
      {errorMessage && (
        <div>
          <p id="error-text">{errorMessage}</p>
        </div>
      )}

      <button id="submit-btn" type="submit">
        Send Message &rarr;
      </button>
    </section>
  );
}

export default Contact;
