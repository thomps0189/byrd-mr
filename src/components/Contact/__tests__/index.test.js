import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Contact from "..";

afterEach(cleanup);

describe("Contact component", () => {
  it("renders", () => {
    render(<Contact />);
  });

  // eslint-disable-next-line jest/no-identical-title
  it("renders", () => {
    const { asFragment } = render(<Contact />);
    expect(asFragment()).toMatchSnapshot();
  });
});
