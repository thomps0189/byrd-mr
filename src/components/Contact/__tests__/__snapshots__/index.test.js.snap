// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component renders 1`] = `
<DocumentFragment>
  <section
    id="form-container"
  >
    <form
      id="contact-form"
    >
      <label
        for="full-name"
        id="form-header"
      >
        Full Name
      </label>
      <input
        id="full-name"
        name="full-name"
        placeholder="Enter your full name"
        type="text"
        value=""
      />
      <label
        for="email"
        id="form-header"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        placeholder="Enter your email"
        type="email"
        value=""
      />
      <label
        for="message"
        id="form-header"
      >
        Message
      </label>
      <input
        id="message"
        name="message"
        placeholder="What are your plans?"
        type="text"
        value=""
      />
    </form>
    <button
      id="submit-btn"
      type="submit"
    >
      Send Message â†’
    </button>
  </section>
</DocumentFragment>
`;
